dot/core/service_test.go

```diff
@@ -453,20 +479,69 @@ func TestService_GetMetadata(t *testing.T) {
 }
 
 func TestService_HandleRuntimeChanges(t *testing.T) {
+	const (
+		currSpecVersion           = uint32(260)
+		updatedSpecVersion        = uint32(262)
+		updateNodeRuntimeWasmPath = "../../tests/polkadotjs_test/test/node_runtime.compact.wasm"
+	)
 	s := NewTestService(t, nil)
-	codeHashBefore := s.codeHash
 
-	testRuntime, err := ioutil.ReadFile(runtime.POLKADOT_RUNTIME_FP)
+	hash := s.blockState.BestBlockHash() // geneisHash
+
+	newBlock1 := &types.Block{
+		Header: &types.Header{
+			ParentHash: hash,
+			Number:     big.NewInt(1),
+			Digest:     types.Digest{utils.NewMockDigestItem(1)}},
+		Body: types.NewBody([]byte("Old Runtime")),
+	}
+
+	newBlockRTUpdate := &types.Block{
+		Header: &types.Header{
+			ParentHash: hash,
+			Number:     big.NewInt(1),
+			Digest:     types.Digest{utils.NewMockDigestItem(2)}},
+		Body: types.NewBody([]byte("Updated Runtime")),
+	}
+
+	ts, err := s.storageState.TrieState(nil) // Pass genesis root
 	require.NoError(t, err)
 
-	ts, err := s.storageState.TrieState(nil)
+	parentRt, err := s.blockState.GetRuntime(&hash)
+	require.NoError(t, err)
+
+	v, err := parentRt.Version()
+	require.NoError(t, err)
+	require.Equal(t, v.SpecVersion(), currSpecVersion)
+
+	bhash1 := newBlock1.Header.Hash()
+	err = s.blockState.HandleRuntimeChanges(ts, parentRt, bhash1)
+	require.NoError(t, err)
```

Comment:
can you check that the runtime for `newBlock1` is the same as the parent runtime?